Library/readline/readline.c

```diff
@@ -0,0 +1,93 @@
+
+//
+// readline.c
+//
+// Copyright (c) 2014 Yorkie Neil <yorkiefixer@gmail.com>
+//
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "readline.h"
+
+readline_t *
+readline_new(char * buffer) {
+  readline_t * rl = (readline_t*) malloc(sizeof(readline_t));
```

Comment:
*NULL is reset vector on Z80 family so cannot harm any code or data that way on Z80.